### PR TITLE
Fix incorrect EOS warning message generated by direct-dispatch type queries

### PR DESCRIPTION
**Pull request by pengzhout**
QD should not expect end-of-stream comes from QEs who is not members of
direct dispatch and should not report warning message.

### CODE REVIEW

**Comment by pengzhout** [comment]
repro steps:
```
1. create table test as select i as c1, 999 as c2 from generate_series(1, 100) i;
2. select * from t1 where c1 =2 order by c1, c2;
3. grep 'Motion' master_log
```

@xiong-gang @eurekaka @yydzero

File: src/backend/cdb/motion/cdbmotion.c [hunk 1]
```diff
@@ -735,6 +735,9 @@ EndMotionLayerNode(MotionLayerState *mlStates, int16 motNodeID, bool flushCommLa
 		{
 			pCSEntry = &pMNEntry->ready_tuple_lists[i];
 
+			if (!pCSEntry->init)
```
Comment (atris) [inline src/backend/cdb/motion/cdbmotion.c]: Please add a comment mentioning same.

Otherwise, +1

Comment (pengzhout) [inline src/backend/cdb/motion/cdbmotion.c]: Thanks, atri, updated

**Comment by yydzero** [comment]
+1

**Comment by xiong-gang** [comment]
+1

**Comment by pengzhout** [comment]
merged and closed

**Pull request closed without merging**